### DISCUSSION TITLE
关于init的使用

### DISCUSSION THREAD

**Discussion by fanhousanbu** [go-atomci/atomci#101] [Show and tell]
#95 由于init导致的单测失败

@sampsonye 提供了一个思路，参考 [The promblems of Golang init function](https://liyafu.com/2021-07-07-the-problems-of-go-init-func/#:~:text=There%20are%20several%20ways%20to%20avoid%20the%20init,need%20to%20initialize%2C%20use%20an%20exported%20init%20function.)  

关于文章的结论：不要用init

1. 我们是否需要考虑这件事；

2. 目前的解决方案，文章中也提供了，即在init中判断执行命令是否带-test，稍后我会调整pr以使internal里的单测能顺利执行

**Comment by sampsonye** [comment]
我的意见是：不要使用init

**Reply by colynn** [reply]
我也是支持不使用init ，　转为具体的函数引用
